### ISSUE TITLE
新手的一个问题，关于数据读入的

### ISSUE THREAD

**Issue by lxtGH** [ycszen/tf_lab#4]
首先感谢楼主分享代码！
有2个问题（tf [官方文档没看懂）](https://www.tensorflow.org/programmers_guide/reading_data)
1，就是 `x_train, y_train = read_and_decode("train_tfrecord", is_train=True)`   返回多少数据(一个样本吗)？比方说一个400M的tf_records 文件， 而它后面跟的
`tf.train.shuffle_batch([distorted_image, label], batch_size=25,num_threads=2, capacity=1000, min_after_dequeue=10)`
每次训练时，这个函数是可以拿到一个batch 给计算图谱吗？ 那训练时每次迭代如何保证数据是随机抽取的？
2，tf.train.start_queue_runners（）和这个有什么关系

**Comment by SWHL** [comment]
1.这是这个函数的官网说明：
"""Creates batches by randomly shuffling tensors.

  This function adds the following to the current `Graph`:

  * A shuffling queue into which tensors from `tensors` are enqueued.
  * A `dequeue_many` operation to create batches from the queue.
  * A `QueueRunner` to `QUEUE_RUNNER` collection, to enqueue the tensors
    from `tensors`.
希望能帮到你
2.这个函数就是"""Starts all queue runners collected in the graph.
不影响其他的

**Comment by alyato** [comment]
@SWHL 
@lxtGH 
谢谢。我现在开始用tf，但是对输入不是很明白。
因为我有自己的图片数据集，就是想加载到tf的mnist中运行一下。
不知道有没有完整的代码，读取数据图片，标签，然后输入到网络中。
再次感谢。

**Comment by lxtGH** [comment]
@alyato 
参考这几个博客，可以考虑先把你图片生成tf-record再去读取。
https://www.cnblogs.com/upright/p/6136265.html
https://blog.csdn.net/qq_16949707/article/details/53483493
或者把你的图片和label 对应地读到python list当中（最好名字格式相同，便于sort）

**Comment by alyato** [comment]
@lxtGH 
谢谢。正在看。
